flake.nix

```diff
@@ -0,0 +1,55 @@
+{
+  description = "sn-bindgen";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
+
+  inputs.sbt.url = "github:zaninime/sbt-derivation/master";
+  inputs.sbt.inputs.nixpkgs.follows = "nixpkgs";
+
+  inputs.systems.url = "github:nix-systems/default";
+
+  outputs = {
+    self,
+    nixpkgs,
+    sbt,
+    systems
+  }:
+    let
+      eachSystem = nixpkgs.lib.genAttrs (import systems);
+    in
+    {
+      packages = eachSystem (system:
+        let pkgs = nixpkgs.legacyPackages.${system}; in
+        {
+          default = sbt.mkSbtDerivation.${system} {
+            pname = "sn-bindgen";
+            version = "0.1.0";
+            src = self;
+            depsSha256 = "sha256-3sgPKN1/0lxSxOtetlWS/NobXIpQ6a7Ygk2GOprYIcg=";
+            buildPhase = ''
+              sbt 'show buildBinary'
+            '';
+            depsWarmupCommand = ''
+              sbt 'update'
+              sbt 'iface/compile ; iface2_12/compile ; iface3/compile'
+              sbt 'ifaceNative/compile ; ifaceNative2_12/compile ; ifaceNative3/compile'
+            '';
```

Comment:
I'm not familiar with Nix at all - what is the logic behind which modules are compiled/updated in this step? What does it do in general?

Reply:
It's a command @coreyoconnor came up with on his own to fetch necessary dependencies before building the package, because in the building stage all network requests won't be allowed.

Nix is a special kind of thing, and you do not have to have a nix code in the repository for it in order to be available for nix users. Nix has a huge context, and I am not really sure if it is a necessarily good thing to introduce some Nix code to the repo :) It's fun though.

I myself have been using NixOS for about 1.5 years.